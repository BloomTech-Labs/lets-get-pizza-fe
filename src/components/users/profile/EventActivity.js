import React from 'react'
import { Feed, Divider } from 'semantic-ui-react'
import { useSelector } from 'react-redux'
import "../MobileStyle.css"
import { useHistory } from 'react-router-dom'

const EventActivity = ({ user, event }) => {
  const curr_user = useSelector(({ user }) => user.username);
  const history = useHistory();
  return (
    <>
      <Feed.Event id="example">
        <Feed.Label image={user.profile_image} />
        <Feed.Content>
          <Feed.Summary>
            <Feed.User
              as="a"
              href={
                user.username !== curr_user
                  ? `/users/${user.username}`
                  : `/users/dash/profile`
              }
            >
              {user.username}
            </Feed.User>{" "}
            {event.response
              ? "was invited to an event at "
              : "posted an event at "}
            {event.business_name}
          </Feed.Summary>
          <br />
          <Feed.Summary
            as="a"
            onClick={() =>
              history.push(`/locations/${event.location_id}/events`)
            }
          >
            {event.title}
          </Feed.Summary>
          <Feed.Extra text>{event.description}</Feed.Extra>
          <Feed.Meta>
            <Feed.Like icon="thumbs up" /> 13 Likes{" "}
          </Feed.Meta>
        </Feed.Content>
      </Feed.Event>
      <Divider />
    </>
  );
};

export default EventActivity;
