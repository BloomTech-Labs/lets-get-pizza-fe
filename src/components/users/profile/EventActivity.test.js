import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { useHistory } from "react-router-dom";
import EventActivity from "./EventActivity";
import {
  findElementById,
  findFirstProp,
} from "../../../utils/reduxTestingFunctions";

const mockStore = configureStore({});

describe("event activity", () => {
  let store;
  let component;

  const event = {
    business_name: "pizza man",
    location_id: 1122,
    description: "this is a test man",
    title: "the pizza place",
  };

  const user = {
    profile_image:
      "https://vignette.wikia.nocookie.net/dragonballfighterz/images/e/ea/Goku_Artwork.png/revision/latest/top-crop/width/360/height/450?cb=20180902173423",
    username: "AnimeCody",
  };

  beforeEach(() => {
    store = mockStore({
      user: user,
    });

    component = renderer.create(
      <Provider store={store}>
        <EventActivity user={user} event={event} />
      </Provider>
    );
  });

  it(" profile image matches state", () => {
    const image = component.toJSON()[0].children[0].children[0].props.src;
    expect(image).toBe(user.profile_image);
  });
  it("should render username from props", () => {
    const ex = component.root.findAllByType("a")[0].props;
    expect(ex.children).toContain(user.username);
  });
});
