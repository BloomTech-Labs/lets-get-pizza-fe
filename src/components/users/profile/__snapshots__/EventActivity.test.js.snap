// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event activity should match snapshot 1`] = `
Array [
  <div
    className="event"
  >
    <div
      className="label"
    >
      <img
        src="https://vignette.wikia.nocookie.net/dragonballfighterz/images/e/ea/Goku_Artwork.png/revision/latest/top-crop/width/360/height/450?cb=20180902173423"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="summary"
      >
        <a
          className="user"
          href="/users/dash/profile"
        >
          AnimeCody
        </a>
         
        posted an event at 
        pizza man
      </div>
      <br />
      <a
        className="summary"
        onClick={[Function]}
      >
        the pizza place
      </a>
      <div
        className="text extra"
      >
        this is a test man
      </div>
      <div
        className="meta"
      >
        <a
          className="like"
        >
          <i
            aria-hidden="true"
            className="thumbs up icon"
            onClick={[Function]}
          />
        </a>
         13 Likes 
         
      </div>
    </div>
  </div>,
  <div
    className="ui divider"
  />,
]
`;
